GPU/Common/SplineCommon.cpp

```diff
@@ -361,16 +383,45 @@ static void _SplinePatchFullQuality(u8 *&dest, u16 *indices, int &count, const S
 	}
 }
 
+template <bool origNrm, bool origCol>
+static inline void SplinePatchFullQualityDispatch3(u8 *&dest, u16 *indices, int &count, const SplinePatchLocal &spatch, u32 origVertType, int quality, int maxVertices) {
+	bool origTc = (origVertType & GE_VTYPE_TC_MASK) != 0;
```

Comment:
The template thing is just an easy microoptimization that probably gains just a few %, I'm sure it's possible to speed it up a larger amount with a bigger rewrite while making it simpler. If you can do that while keeping the new code readable I'll happily merge it.